modules/dnn/src/dnn.cpp

```diff
@@ -2225,7 +2225,9 @@ struct Net::Impl : public detail::NetImplBase
 
                     auto ieInpNode = inputNodes[i].dynamicCast<InfEngineNgraphNode>();
                     CV_Assert(oid < ieInpNode->node->get_output_size());
-#if INF_ENGINE_VER_MAJOR_GT(INF_ENGINE_RELEASE_2020_3)
+#if INF_ENGINE_VER_MAJOR_GT(INF_ENGINE_RELEASE_2020_4)
+                    inputNodes[i] = Ptr<BackendNode>(new InfEngineNgraphNode(ieInpNode->node));
```

Comment:
Seems like this method does not depend on output id (`oid`)

Reply:
Yes, at the current moment this method uses node from ieImplNode.
But it can be incorrect if this node has several outputs. In this case it is better to use `Output<Node>` instead of `std::shared_prt<Node>`